### PR TITLE
functions.js file created

### PR DESCRIPTION
**Pull request by ljmcdade**
created branch and file

### CODE REVIEW

**Comment by ljmcdade** [comment]
question 1 sorted an array by largest string

**Comment by ljmcdade** [comment]
question 2

created objects, sorted by longer name, sorted alphabetically by email

**Comment by ljmcdade** [comment]
question 3 used the sqrt function to create a squaring function

**Comment by ljmcdade** [comment]
question 5

create function that returns a function

File: functions.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+//     }
+//     if (a[1].toUpperCase() > b[1].toUpperCase()) {
+//         return 1;
+//     }
+//     return 0;
+// }
+//     return arrayName.sort(compare);
+// }
+// console.log(alphabeticalEmail(arrayName));
+
```
Comment (noratop) [inline functions.js]: @ljmcdade As we saw briefly together yesterday, it seems you have some misunderstanding with the function `sort`. We should review it together today.

File: functions.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+// // roots is now [1, 2, 3], numbers is still [1, 4, 9]
+
+// var numbers = [1, 4, 9];
+// var squares = numbers.map(square);
+// // roots is now [1, 2, 3], numbers is still [1, 4, 9]
+
+// function square(n){
+//     return n*n;
+// }
+// console.log(squares);
```
Comment (noratop) [inline functions.js]: @ljmcdade :+1:

File: functions.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+//  }
+
+// function squareNum(currentObject) {
+//     currentObject.num = currentObject.num * currentObject.num;
+//     return currentObject;
+// }
+
+// var squares = arrayOfObject.map(squareNum);
+
+// console.log(squares);
```
Comment (noratop) [inline functions.js]: @ljmcdade :+1:

File: functions.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    }
+}
+
+var adding = operationMaker("add");
+var sum = adding(4, 6);
+
+var multiplying = operationMaker("multiply");
+var product = multiplying(5, 7);
+
+console.log(sum, product);
```
Comment (noratop) [inline functions.js]: @ljmcdade I believe this exercise is not done? don't hesitate to get in touch for help!

**Pull request closed without merging**